unskript-ctl/unskript_ctl_database.py

```diff
@@ -108,7 +109,25 @@ def update(self, **kwargs):
             data = kwargs.get('data')
         with self.db.transaction() as connection:
             root = connection.root()
-            old_data = root[self.collection_name]
+            stored_data = root.get(self.collection_name, {})
+            if isinstance(stored_data, bytes):
+                self.logger.info("Found database to be using old schema, will migrate to new format")
+                try:
+                    decompressed_data = zlib.decompress(stored_data)
+                    old_data = json.loads(decompressed_data.decode('utf-8'))
+                except zlib.error as e:
+                    self.logger.error(f"Error decompressing data: {e}")
+                    return False
+                except json.JSONDecodeError as e:
+                    self.logger.error(f"Error decoding JSON data: {e}")
+                    return False
+                finally:
+                    self.logger.info("DB schema migrated successfully")
```

Comment:
Is it the right log? Is it actually migrated? you havent put back the new schema to the disk i think?

Reply:
Done, had missed adding the SCHEMA_VERSION.